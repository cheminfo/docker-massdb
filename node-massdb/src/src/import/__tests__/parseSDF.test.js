'use strict';

let join = require('path').join;

let parseSDF = require('../parseSDF');

test('test SDF parsing', async () => {
  let file = join(__dirname, 'data/nist.sdf');
  let result = await parseSDF(file);
  expect(Object.keys(result)).toHaveLength(2);
  expect(result).toMatchSnapshot();
});
