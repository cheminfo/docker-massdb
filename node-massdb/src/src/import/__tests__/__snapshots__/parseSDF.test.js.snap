// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test SDF parsing 1`] = `
Object {
  "Deuterium": Object {
    "molfile": "Deuterium
  -NISTMS-12099911132D 1   1.0         0.0          

  2  1  0  0  0  0  0  0  0  0  0
   -0.4060    0.6090    0.0000 H   1  0  0     0  0  0  0  0  0
    0.4060    0.6090    0.0000 H   1  0  0     0  0  0  0  0  0
  1  2  1  0  0  0  0
$$$$
",
    "name": "Deuterium",
  },
  "Hydrogen": Object {
    "molfile": "Hydrogen
  -NISTMS-12099911132D 1   1.0         0.0          

  2  1  0  0  0  0  0  0  0  0  0
   -0.4060    0.6090    0.0000 H   0  0  0     0  0  0  0  0  0
    0.4060    0.6090    0.0000 H   0  0  0     0  0  0  0  0  0
  1  2  1  0  0  0  0
$$$$
",
    "name": "Hydrogen",
  },
}
`;
